src/main/java/net/mcreator/workspace/references/ResourceReference.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2023, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.workspace.references;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target(ElementType.FIELD) @Retention(RetentionPolicy.RUNTIME) public @interface ResourceReference {
+	String value();

```

Comment:
probably type(), not value.

Reply:
`value()` allows to omit the property name, which I think should be intuitive here, isn't it?

Reply:
Wait, this is not `@TestureReference` :man_facepalming: Should I rename it?

Reply:
Wait, this is not `@TextureReference` :man_facepalming: Should I rename it?

Reply:
Afaik if there is only one method, it can have any name

Reply:
Just checked, Java requires the only field to be named `value`.